pkg/declarativeconfig/transform/role.go

```diff
@@ -0,0 +1,36 @@
+package transform
+
+import (
+	"reflect"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/declarativeconfig"
+	"github.com/stackrox/rox/pkg/errox"
+)
+
+var _ Transformer = (*roleTransform)(nil)
+
+type roleTransform struct{}
+
+func newRoleTransform() *roleTransform {
+	return &roleTransform{}
+}
+
+func (r *roleTransform) Transform(configuration declarativeconfig.Configuration) (map[reflect.Type][]proto.Message, error) {
+	roleConfig, ok := configuration.(*declarativeconfig.Role)
+	if !ok {
+		return nil, errox.InvalidArgs.Newf("invalid configuration type received for role: %T", configuration)
+	}
+
+	roleProto := &storage.Role{
+		Name:            roleConfig.Name,
+		Description:     roleConfig.Description,
+		PermissionSetId: roleConfig.PermissionSet,
```

Comment:
That does not seem right - declarative configuration contains names of permission set and access scope, not ids.